scribus/text/screenpainter.cpp

```diff
@@ -125,6 +126,8 @@ void ScreenPainter::drawGlyph(const GlyphCluster& gc)
 		{
 			outline = m_item->doc()->symTab.copy();
 			chma4.translate(gc.width() - fontSize() * 0.7, -fontSize() * gc.scaleV() * 0.5);
+            if(gc.hasFlag(ScLayout_RightToLeft))
+                    flipHorizontal(chma4);
```

Comment:
Please be sure to respect the indentation. The easiest way to do it in Qtcreator select the 4 lines and press ctrl+i